webviz_subsurface/_utils/fanchart_plotting.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any, Dict, List, Optional
+from dataclasses import dataclass, field
+from enum import Enum
+
+import numpy as np
+
+from .colors import hex_to_rgba
+
+
+@dataclass
+class FanchartData:
+    """
+    Dataclass defining fanchart data utilized in creation of statistical fanchart traces
+
+    Attributes:
+    samples - Common sample point list for each following value list.
+    mean    - 1D np.array of mean value data.
+    maximum - 1D np.array of maximum value data.
+    low     - 1D np.array of low percentile value data.
+    high    - 1D np.array of high percentile value data.
+    minimum - 1D np.array of minimum value data.
+    """
+
+    samples: list = field(default_factory=list)
+    mean: Optional[np.ndarray] = None
+    maximum: Optional[np.ndarray] = None
+    low: Optional[np.ndarray] = None
+    high: Optional[np.ndarray] = None
+    minimum: Optional[np.ndarray] = None
+
+
+class TraceDirection(Enum):
+    HORIZONTAL = 1
+    VERTICAL = 2
+
+
+def validate_fanchart_data(data: FanchartData) -> None:
+    """
+    Validation of fanchart data
+
+    Ensure equal length of all statistical fanchart data lists and x-axis data list
+
+    Raise ValueError if lengths are unequal
+    """
+    if len(data.samples) <= 0:
+        raise ValueError("Empty x-axis data list in FanchartData")
+    if data.mean is not None and len(data.samples) != len(data.mean):
+        raise ValueError(
+            "Invalid fanchart mean value data length. len(data.samples) != len(data.mean)"
+        )
+    if data.maximum is not None and len(data.samples) != len(data.maximum):
+        raise ValueError(
+            "Invalid fanchart maximum value data length. len(data.samples) != len(data.maximum)"
+        )
+    if data.low is not None and len(data.samples) != len(data.low):
+        raise ValueError(
+            "Invalid fanchart low percentile value data length. len(data.samples) != len(data.low)"
+        )
+    if data.high is not None and len(data.samples) != len(data.high):
+        raise ValueError(
+            "Invalid fanchart high percentile value data length. "
+            "len(data.samples) != len(data.high)"
+        )
+    if data.minimum is not None and len(data.samples) != len(data.minimum):
+        raise ValueError(
+            "Invalid fanchart minimum value data length. len(data.samples) != len(data.minimum)"
+        )
+
+
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-locals
+def get_fanchart_traces(
+    data: FanchartData,
+    color: str,
+    legend_group: str,
+    line_shape: str = "linear",
+    xaxis: str = "x",
+    yaxis: str = "y",
+    low_percentile_name: str = "P90",
+    high_percentile_name: str = "P10",
```

Comment:
Consider to drop the default, to make sure that those using the feature actually think about it.
```suggestion
    low_percentile_name: str,
    high_percentile_name: str,
```

Reply:
Handled in refactoring in last commit